lib/core.ps1

```diff
@@ -1038,22 +1064,30 @@ function shim($path, $global, $name, $arg) {
 
         warn_on_overwrite $shim $path
         @(
-            "#!/bin/sh",
+            '#!/bin/sh',
             "# $resolved_path",
             "python.exe `"$resolved_path`" $arg `"$@`""
         ) -join "`n" | Out-UTF8File $shim -NoNewLine
     } else {
         warn_on_overwrite "$shim.cmd" $path
         @(
             "@rem $resolved_path",
-            "@bash `"$resolved_path`" $arg %*"
+            "@bash `"`$(wslpath -u '$resolved_path')`" $arg %* 2>nul",
+            '@if %errorlevel% neq 0 (',
+            "  @bash `"`$(cygpath -u '$resolved_path')`" $arg %* 2>nul",
+            ')'
         ) -join "`r`n" | Out-UTF8File "$shim.cmd"
 
         warn_on_overwrite $shim $path
         @(
-            "#!/bin/sh",
+            '#!/bin/sh',
             "# $resolved_path",
-            "`"$resolved_path`" $arg `"$@`""
+            "if [ `$WSL_INTEROP ]",
```

Comment:
```suggestion
            "if [ `$WSL_INTEROP` ]",
```
Doesn't a closing <tt>`</tt> belong here?

Reply:
The ` is escaping the $.